src/Spice86.Core/Emulator/InterruptHandlers/Input/Keyboard/KeyboardInt16Handler.cs

```diff
@@ -60,80 +55,92 @@ private void Unsupported(int operation) {
     public override byte VectorNumber => 0x16;
 
     /// <summary>
-    ///     Emits a minimal INT 16h handler stub into guest RAM that handles AH=00h (GetKeystroke) in-place
-    ///     and dispatches all other AH values back to the C# handler via a callback.
+    /// Emits a minimal INT 16h handler stub into guest RAM.
     /// </summary>
     /// <remarks><![CDATA[
-    ///     Assembled control flow (labels for readability):
-    ///     L_HANDLER:
+    /// Handles AH=00h (Get Keystroke) locally with a busy-wait loop driven by a callback that sets ZF,
+    /// and dispatches all other AH values back to the C# handler.
+    /// 
+    /// Pseudo-assembly (labels for readability):
+    /// 
+    ///   L_HANDLER:
     ///     cmp ah, 00h
     ///     jz  L_AH00
     /// 
-    ///     L_DEFAULT:
-    ///     int 0A4h   ; DefaultDispatch -> calls C# Run()
+    ///   L_DEFAULT:
+    ///     int cbDefaultDispatch   ; calls C# Run() to handle AH != 00h
     ///     iret
     /// 
-    ///     L_AH00:
-    ///     int 0A0h   ; CallbackHasKey -> sets ZF=0 if a key is present
-    ///     jnz have_key
-    ///     int 09h    ; invoke hardware keyboard ISR (IRQ1) to fetch a key
-    ///     jmp short L_AH00
-    /// 
-    ///     have_key:
-    ///     int 0A1h   ; CallbackDequeueAndSetAx -> AX = (scan<<8 | ascii)
+    ///   L_AH00:
+    ///   L_WAIT:
+    ///     int cbGetKeystrokeStatus ; sets AX and ZF
+    ///     jz  L_WAIT               ; ZF=1 => no key available, keep spinning with IF enabled
+    ///     cli
+    ///     int cbDequeueAndSetAx    ; AX = (scan<<8 | ascii), consumes the key
+    ///     sti
     ///     iret
     /// 
-    ///     Callback vector mapping:
-    ///     - 0xA0 => CallbackHasKey
-    ///     - 0xA1 => CallbackDequeueAndSetAx
-    ///     - 0xA4 => Run (default C# dispatcher for unsupported AH values)
-    ///     Notes:
-    ///     - The short jumps are encoded to skip over the DefaultDispatch+IRET and to loop back while waiting.
-    ///     - This keeps the common AH=00h path fast in RAM while preserving behavior for other functions.
+    /// Notes:
+    /// - Interrupts remain enabled during the wait loop so IRQ1 can populate the BIOS buffer.
+    /// - A short JZ right after the CMP skips exactly the default-dispatch INT+IRET sequence.
     /// ]]></remarks>
     /// <param name="memoryAsmWriter">Helper used to write machine code and callbacks into guest memory.</param>
     /// <returns>The segment:offset address where the handler stub was emitted.</returns>
     public override SegmentedAddress WriteAssemblyInRam(MemoryAsmWriter memoryAsmWriter) {
-        // Only AH=00 (GetKeystroke) is implemented in the in-RAM handler.
-        // All other AH values jump to the default C# dispatcher via callback.
-
         SegmentedAddress handlerAddress = memoryAsmWriter.CurrentAddress;
 
+        // Ensure IF is set on entry (like PC XT BIOS does)
```

Comment:
"Ensure IF is set on entry to let keyboard interrupt execute" (explains intent ?)